### ISSUE TITLE
Signal Desktop faulty version potentially causing data loss

### ISSUE THREAD

**Issue by norpol** [NixOS/nixpkgs#108772]
**Describe the bug**
Just lost on two different NixOS machine my Signal Desktop history because of this faulty update...
I don't know, probably doesn't make sense for people to downgrade who are already upgraded but for others it dataloss

https://github.com/signalapp/Signal-Desktop/issues/4513

**To Reproduce**

1. Upgrade Signal Desktop
2. Restart Signal Desktop
3. Asks you to delete db, if you interact in a wrong way it will delete your data

**Notify maintainers**

@primeos (@ixmatus and @equirosa)

**Comment by primeos** [comment]
> Just lost on two different NixOS machine my Signal Desktop history because of this faulty update...

Sorry to hear that. I recently lost my history too... It's really frustrating!

> I don't know, probably doesn't make sense for people to downgrade who are already upgraded but for others it dataloss

Yeah, unfortunately I don't think there's anything we can do about it (as downstream package maintainers).
AFAIK the source of the bug is still unknown, it doesn't seem to affect all users, and all releases have a limited lifetime ("expire" 90 days after the release, see my comment in `pkgs/applications/networking/instant-messengers/signal-desktop/default.nix`).

> I hate Signals way of dealing with long term conversations and history archival, it sucks - there are no proper ways of dealing with this. @moxie0

Thanks for saying that! This really bothered me too... I just didn't say anything because I was too angry to keep it nice enough.
(I raged a bit here https://social.primeos.dev/notice/A22N5eEVCDrkxyDmwy but after I realized that one cannot even import old backups I had to take a break.)

**Comment by RaitoBezarius** [comment]
I just got hit by a similar issue by moving from Archlinux (encrypted database) to NixOS (non encrypted database).

After many things, the easiest fix to ensure proper functionalities IMHO is installing:
```
pkgs.signal-desktop.override {
  gtk3 = pkgs.gtk3.override { trackerSupport = false; };
}
```

This will disable tracker3 and prevent SQLite to be loaded in the process which will replace SQLCipher and cause the SQLite3 API to be the original one (i.e. non-encrypted) and cause many issues down the road I guess.

That might not fix everything though :upside_down_face:

**Comment by primeos** [comment]
FYI, in case anyone has feedback/suggestions: We might merge #117555 soon and it should cause another "Database startup error: Error: SQLITE_NOTADB: file is not a database" error (which is really unfortunate but not easily avoidable).

**Comment by primeos** [comment]
Heads-up: I've just merged #122926 to hopefully resolve this once and for all and the Python wrapper script should take care of the database re-encryption when launching Signal-Desktop as usual but the user experience isn't that great and you probably want to backup `~/.config/Signal/` first in case the script fails unexpectedly.

**Comment by norpol** [comment]
Thank you, I just upgraded Signal on my computer and the re-encryption did work as intended
